src/Annotations/Serializer/AnnotationNormalizer.php

```diff
@@ -0,0 +1,157 @@
+
+        $data = array_filter([
+            'id' => $object->getId(),
+            'access' => ($object->getPermissions()->getRead() === Annotation::PUBLIC_GROUP) ? 'public' : 'restricted',
+            'content' => $content,
+            'parents' => $object->getReferences(),
+            'created' => $object->getCreatedDate()->format(ApiSdk::DATE_FORMAT),
+            'updated' => $object->getUpdatedDate()->format(ApiSdk::DATE_FORMAT),
+            'document' => [
+                'title' => $object->getDocument()->getTitle(),
+                'uri' => $object->getUri(),
+            ],
+        ]) + ['parents' => []];
+        if ($data['created'] === $data['updated']) {
+            unset($data['updated']);
+        }
+        if ($object->getTarget()->getSelector()) {
+            $data['highlight'] = $object->getTarget()->getSelector()->getTextQuote()->getExact();
+        }
+        if (empty($data['highlight']) && empty($data['content'])) {
+            $this->logger->warning(sprintf('Annotation detected without highligt or content (ID: %s)', $data['id']), ['annotation' => $data]);
+            $data['content'] = self::CANNOT_RENDER_CONTENT_COPY;
+        }
+
+        return $data;
+    }
+
+    private function processText(string $text) : array
+    {
+        $blocks = $this->docParser->parse($text)->children();
+        $data = [];
+
+        foreach ($blocks as $block) {
+            $rendered = $this->htmlRenderer->renderBlock($block);
+            if (empty($rendered)) {
+                continue;
+            }
+
+            switch (true) {
+                case $block instanceof Element\ThematicBreak:
+                    break;
+                case $block instanceof Element\ListBlock:
+                    $data[] = $this->processListBlock($block);
+                    break;
+                case $block instanceof Element\BlockQuote:
+                    $data[] = [
+                        'type' => 'quote',
+                        'text' => [
+                            [
+                                'type' => 'paragraph',
+                                'text' => $rendered,
+                            ],
+                        ],
+                    ];
+                    break;
+                case $block instanceof Element\HtmlBlock:
+                case $block instanceof CommonMark\Block\Element\Latex:
+                case $block instanceof CommonMark\Block\Element\MathML:
+                case $block instanceof Element\Paragraph:
+                    $data[] = [
+                        'type' => 'paragraph',
+                        'text' => $rendered,
+                    ];
+                    break;
+                case $block instanceof Element\FencedCode:
+                case $block instanceof Element\IndentedCode:
+                    $data[] = [
+                        'type' => 'code',
+                        'code' => $rendered,
+                    ];
+                    break;
```

Comment:
no `default`? Either a default action (or a `throw SomeException` if it's never meant to get there) would be useful